types/const_test.go

```diff
@@ -279,12 +279,12 @@ func (s *testMySQLConstSuite) TestPadCharToFullLengthMode(c *C) {
 	tk.MustExec("insert into t1 values ('xy');")
 	tk.MustExec("set sql_mode='';")
 	r = tk.MustQuery(`SELECT a='xy        ', char_length(a) FROM t1;`)
-	r.Check(testkit.Rows("0 2"))
+	r.Check(testkit.Rows("1 2"))
 	r = tk.MustQuery(`SELECT count(*) FROM t1 WHERE a='xy        ';`)
-	r.Check(testkit.Rows("0"))
+	r.Check(testkit.Rows("1"))
 	tk.MustExec("set sql_mode='PAD_CHAR_TO_FULL_LENGTH';")
 	r = tk.MustQuery(`SELECT a='xy        ', char_length(a) FROM t1;`)
-	r.Check(testkit.Rows("0 2"))
+	r.Check(testkit.Rows("1 2"))
```

Comment:
Actually, MySQL returns `1 10` here.
We may add a comment as `FIXME` here.

Reply:
In my mysql 5.7.28
```
mysql> create table t1 (a varchar(10));
Query OK, 0 rows affected (0.01 sec)

mysql> insert into t1 values ('xy');
Query OK, 1 row affected (0.01 sec)

mysql> set sql_mode='PAD_CHAR_TO_FULL_LENGTH';
Query OK, 0 rows affected (0.00 sec)

mysql> SELECT a='xy        ', char_length(a) FROM t1;
+----------------+----------------+
| a='xy        ' | char_length(a) |
+----------------+----------------+
|              1 |              2 |
+----------------+----------------+
1 row in set (0.00 sec)
```

Reply:
my fault here